### DISCUSSION TITLE
Warning message

### DISCUSSION THREAD

**Discussion by silasalvescosta** [jamesjiadazhan/dietaryindex#127] [Q&A] [answered]
Hi, all
I'm trying to calculate HEI2015 for two NHANES 11-12 and 13-14. I think that I need to calculate it for each database separately and merge them after. When I start running the following code for NHANES 11-12, a warning message appears:

```
FPED_PATH = "C:/fped_dr1tot_1112.sas7bdat"
NUTRIENT_PATH = "C:/DR1TOT_G.XPT"
DEMO_PATH = "C:/DEMO_G.XPT"

HEI2015_NHANES_FPED(FPED_PATH, NUTRIENT_PATH, DEMO_PATH)

#Use the example data
data("NHANES_1114")
HEI2015_NHANES_FPED(NHANES_1114$FPED, NHANES_1114$NUTRIENT, NHANES_1114$DEMO)

Warning message:
In data("NHANES_1114") : data set ‘NHANES_1114’ not found
```

**The questions are:** 
The `data("NHANES_1114")` means the original database with the other information/ complete database/ merged?
Where do I need to have previously saved the original database ("NHANES_1114")?
I use the database as data.frame but the message persists.

Can I use the combined database?

**Comment by jamesjiadazhan** [comment, marked as answer]
you're right about calculating each and then merge them together. Actually, the `data()` function means you pull the example data from the dietaryindex package for your to quickly try the function. The package only has NHANES_20172018 data. It's why there is a warning because NHANES_1114 doesn't exist at all.

In R, you can store a result by using "=" to create a new variable, as shown below (the FPED, NUTRIENT, OR DEMO files name may not be correct, but the ideas are there). Also, use full_join to join 2 results if you would like to preserve all rows:



```
#1112 data
FPED_1112= read_sas("C:/fped_dr1tot_1112.sas7bdat") 
NUTRIENT_1112 = read_xpt("C:/DR1TOT_G.XPT") %>%
    filter(DR1DRSTZ == 1)
DEMO_1112 = read_xpt("C:/DEMO_G.XPT") %>%
    filter(RIDAGEYR >= 2)
HEI2015_NHANES_1112 = HEI2015_NHANES_FPED(FPED_1112, NUTRIENT_1112, DEMO_1112)

#1314 data
FPED_1314= read_sas("C:/fped_dr1tot_1314.sas7bdat") 
NUTRIENT_1314 = read_xpt("C:/DR1TOT_G.XPT") %>%
    filter(DR1DRSTZ == 1)
DEMO_1314 = read_xpt("C:/DEMO_G.XPT") %>%
    filter(RIDAGEYR >= 2)

HEI2015_NHANES_1314 = HEI2015_NHANES_FPED(FPED_1314, NUTRIENT_1314, DEMO_1314)

# Merge 1112 and 1314
HEI2015_NHANES_11121314= full_join(HEI2015_NHANES_1112, HEI2015_NHANES_1314, by=c("SEQN" = "SEQN"))
```

Note, if you would like to finally save the results in your computer, you should convert it to dataframe and then save it as csv, something like below.

```
HEI2015_NHANES_11121314_df = as.data.frame(HEI2015_NHANES_11121314)
# Save the result on your computer
readr::write_csv(HEI2015_NHANES_11121314_df, "/your_output_file_location/HEI2015_NHANES_11121314_df.csv")
```

Let me know if you have other questions. Glad to help. If you find the dietaryindex package helpful, please consider giving it a Star on GitHub. It helps recognize my work. Also, if you publish the paper, please cite the package as GitHub “dietaryindex”. The package will be written as a paper soon.

**Reply by silasalvescosta** [reply]
Thank you so much!
You are helping me solve a big problem.

**Reply by jamesjiadazhan** [reply]
You're very welcome.

By the way, I think another approach that merges all DEMO, NUTRIENT, and FPED files together ahead and put them into the HEI2015_NHANES might work too. It may be more intuitive and easy to process data. 

To do this, the below steps could be followed:

```
#1112 data
FPED_1112= read_sas("C:/fped_dr1tot_1112.sas7bdat") 
NUTRIENT_1112 = read_xpt("C:/DR1TOT_G.XPT") %>%
    filter(DR1DRSTZ == 1)
DEMO_1112 = read_xpt("C:/DEMO_G.XPT") %>%
    filter(RIDAGEYR >= 2)
HEI2015_NHANES_1112 = HEI2015_NHANES_FPED(FPED_1112, NUTRIENT_1112, DEMO_1112)

#1314 data
FPED_1314= read_sas("C:/fped_dr1tot_1314.sas7bdat") 
NUTRIENT_1314 = read_xpt("C:/DR1TOT_G.XPT") %>%
    filter(DR1DRSTZ == 1)
DEMO_1314 = read_xpt("C:/DEMO_G.XPT") %>%
    filter(RIDAGEYR >= 2)

# merge 1112 data with 1314 data
FPED_1112_1314 = full_join(FPED_1112, FPED_1314, by=c("SEQN" = "SEQN"))
NUTRIENT_1112_1314 = full_join(NUTRIENT_1112, NUTRIENT_1314, by=c("SEQN" = "SEQN"))
DEMO_1112_1314 = full_join(DEMO_1112, DEMO_1314, by=c("SEQN" = "SEQN"))

# HEI2015 result
HEI2015_NHANES_11121314 =  HEI2015_NHANES_FPED(FPED_1112_1314, NUTRIENT_1112_1314, DEMO_1112_1314)

```
